text/XXXX-replace_publish_clone_with_push_pull.md

```diff
@@ -0,0 +1,258 @@
+
+Automatic pulling requires us to make a decision on this, and I think it's the right choice is to assume network access by default. We can take a cue from package managers, which also have a namespace, and assume network acess, automatically making network requests when the user seeks datasets they don't have. `npm install some_package` downloads from the web. Both npm and qri have name resolution systems that make this process possible. From a UX perspective npm lets me _ignore_ the network and focus on writing software.
+
+We've struggled with the word "network" because unlike npm, qri also has a peer-2-peer system, but I think the simple answer to needing assets that may only be available in a p2p context are to show an error asking users to run `qri connect`.
+
+I want to live in a world where I can punch a command into `qri sql` that references a dataset & don't have. Qri should figure out what I'm talking about, go find that dataset, pull the latest version, and run the query. The automatic pulling characteristics of `checkout` would get us there.
+
+### Apply `acceptSizeMax` to a history instead of single verisons
+When pushing a dataset, the receiver is in charge of what they will accept. A remote can reject a push request for any reason. One of the primary reasons for rejecting a push is do to size constraints. Qri has a configuration value `remote.acceptSizeMax` that currently polices the maximium size of a dataset version. Any dataset version that exeeds this value is rejected by default.
+
+This RFC proposes changing the `remote.acceptSizeMax` value in configuration to apply to the on-disk size of _the entire dataset history_. As an example: if `acceptSizeMax` is `300Mb` that quota can be filled with one version that is `300Mb` in size, three totally unique `100Mb` versions, or any other combination where the deduplicated size of all versions is below the maximum threshold. 
+
+We say totally unique versions because qri de-duplicates versioned data. A block of dataset data may be used in any number of versions, and should only count towards the size threshold once.
+
+### Retention Strategies
+This RFC introduces the term _retention strategy_ to describe the approach a peer takes to pruning versions when `acceptSizeMax` is exceeded. A few potential strategies, and their names might be:
```

Comment:
I'm still concerned with what the "default" should be here. If we have too many of any specific type it can negatively impact the network health.
- None: too many of these lead to an only latest N that fit are really available until storage is saturated at which point we have no new network updates (highly susceptible to `acceptSizeMax` and the specific dataset at hand)
- Archival:  too many of these lead to only the earliest N are available to the network (also think these are much more storage heavy but great for long term network health)
- Rolling Window: same as none only that we have just the latest N (I'm especially conerned with edge cases like if a dataset size is near the limit which means we only keep the latest copy)

We should have a strategy on how to balance this out. Think rolling window is the best default as it can be just left running and at least have the latest with archival nodes existing to balance things out. We should definitely investigate the best rations for network health and monitor that closely (metrics are important here). 

As I see it - users are typically `rolling-window`, remotes are `none` and registries are `archival`